### PR TITLE
Port yuzu-emu/yuzu#2188 and yuzu-emu/yuzu#2190: various minor code refactoring changes

### PR DESCRIPTION
**Pull request by FearlessTobi**
See yuzu-emu/yuzu#2188 and yuzu-emu/yuzu#2190 for more details.

I strongly advise to review commit-by-commit.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4681)

### CODE REVIEW

File: src/video_core/renderer_opengl/renderer_opengl.cpp [hunk 1]
```diff
@@ -559,7 +556,11 @@ Core::System::ResultStatus RendererOpenGL::Init() {
         glDebugMessageCallback(DebugHandler, nullptr);
     }
 
-    AddTelemetryFields();
+    const char* const gl_version{reinterpret_cast<char const*>(glGetString(GL_VERSION))};
+    const char* const gpu_vendor{reinterpret_cast<char const*>(glGetString(GL_VENDOR))};
+    const char* const gpu_model{reinterpret_cast<char const*>(glGetString(GL_RENDERER))};
```
Comment (wwylele) [inline src/video_core/renderer_opengl/renderer_opengl.cpp]: Why are these `glGetString` lines moved out of the `AddTelemetryFields` function?

Comment (FearlessTobi) [reply src/video_core/renderer_opengl/renderer_opengl.cpp]: Because `gpu_vendor` is used as variable in line 565. Do you have any better idea?

Comment (wwylele) [reply src/video_core/renderer_opengl/renderer_opengl.cpp]: @FearlessTobi I suggest reverting this commit as this amendment defeats the idea of the original commit.

Comment (FearlessTobi) [reply src/video_core/renderer_opengl/renderer_opengl.cpp]: Dropped the commit.

**Comment by zhaowenlan1779** [comment]
CI failure seems related

**Comment by FearlessTobi** [comment]
Fixed.

**Comment by FearlessTobi** [comment]
Will merge soon if no more reviews.

**Pull request merged**